infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationStreamNameTableManager.java

```diff
@@ -86,10 +104,12 @@ public Set<String> getStreamsToReplicate() {
             openExistingTable(LOG_REPLICATION_PLUGIN_VERSION_TABLE);
             openExistingTable(LOG_REPLICATION_STREAMS_NAME_TABLE);
             if (!tableVersionMatchesPlugin()) {
-                // delete the tables and recreate them
+                // Case: Upgrade - delete the tables and recreate them
                 deleteExistingStreamNameAndVersionTables();
                 createStreamNameAndVersionTables(
                     logReplicationConfigAdapter.fetchStreamsToReplicate());
+                // Persist upgrade flag so a snapshot-sync is enforced upon negotiation
+                updateUpgradeFlag(true);
             }
         } else {
             // If any 1 of the 2 tables does not exist, delete and recreate them both as they may have been corrupted.
```

Comment:
should we also trigger forced full sync in this scenario?

Reply:
We had an offline discussion and thought this should be addressed in a follow-up PR